### PR TITLE
Mailroom searching / filtering / flow start preview

### PR DESCRIPTION
**Pull request by nicpottier**
Supercedes: https://github.com/nyaruka/rapidpro/pull/2654

### CODE REVIEW

**Comment by nicpottier** [comment]
@rowanseymour this is probably as good a checkpoint to ship than anything.

Next up:
 - omnibox searching
 - dynamic group creation / updating
 - dynamic group reevalation for a set of contacts

I *think* then we'll be able to rip out all the ES stuff.

File: temba/mailroom/tests.py [hunk 1]
```diff
@@ -24,18 +24,58 @@ def test_version(self):
 
         self.assertEqual("5.3.4", version)
 
-    def test_flow_migrate(self):
```
Comment (rowanseymour) [inline temba/mailroom/tests.py]: did this unit test get deleted by accident?

File: temba/utils/models.py [hunk 2]
```diff
@@ -26,31 +26,64 @@ def patch_queryset_count(qs, function):
     qs.count = types.MethodType(lambda s: function(), qs)
 
 
-class ProxyQuerySet(object):
+class IDSliceQuerySet(models.query.RawQuerySet):
```
Comment (rowanseymour) [inline temba/utils/models.py]: well this is clever..

Comment (nicpottier) [reply temba/utils/models.py]: Time will tell if its TOO clever, but its a hell of a lot simpler and easier to grok than tom-foolery with both the queryset and paginator.

File: temba/mailroom/tests.py (hunk 1 above)
Comment (nicpottier) [reply temba/mailroom/tests.py]: What in the world, yes apparently so.

**Comment by nicpottier** [comment]
Note that I'm not in love with all parts of this (for example, maybe its worth wrapping the mailroom calls, not sure?) but figure once it is all out I can revisit if there are further simplifications when I see all the call patterns.

File: temba/utils/models.py (hunk 2 above)
Comment (rowanseymour) [reply temba/utils/models.py]: for sure.. had a quick google for other solutions to this but couldn't see anything that would work with slicing/paging

**Pull request merged**